### Importing libraries

In [2]:
#import standard libraries
import pandas as pd
import numpy as np
import os

In [3]:
#creating path
pathk=r'/Users/keerthiputti/Downloads/Achievement 6/cancer incidents/02 Data'

In [4]:
print(pathk)

/Users/keerthiputti/Downloads/Achievement 6/cancer incidents/02 Data


## Plan for data set
I have two data sets for cancer incidents. 2018 and 2019 data.
I will be cleanign them separately and append the data into one set.

In [5]:
#importing all data
df_2019=pd.read_csv(os.path.join(pathk,'Original Data','2019_incidents.csv'),index_col=False)
df_2018=pd.read_csv(os.path.join(pathk,'Original Data','2018_incidents.csv'),index_col=False)


## Cleaning 2019 data

In [6]:
df_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12240 entries, 0 to 12239
Data columns (total 14 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Notes                      62 non-null     object 
 1   Leading Cancer Sites       12178 non-null  object 
 2   Leading Cancer Sites Code  12178 non-null  object 
 3   States                     12178 non-null  object 
 4   States Code                12178 non-null  float64
 5   Year                       12178 non-null  float64
 6   Year Code                  12178 non-null  float64
 7   Age Groups                 12178 non-null  object 
 8   Age Groups Code            12178 non-null  object 
 9   Sex                        12178 non-null  object 
 10  Sex Code                   12178 non-null  object 
 11  Count                      12178 non-null  float64
 12  Population                 12178 non-null  float64
 13  Crude Rate                 12178 non-null  flo

In [7]:
df_2019.shape

(12240, 14)

In [8]:
df_2019.head(100)

Notes            Leading Cancer Sites Leading Cancer Sites Code    States  \
0    NaN  Brain and Other Nervous System               31010-31040   Alabama   
1    NaN  Brain and Other Nervous System               31010-31040   Alabama   
2    NaN  Brain and Other Nervous System               31010-31040   Alabama   
3    NaN  Brain and Other Nervous System               31010-31040   Alabama   
4    NaN  Brain and Other Nervous System               31010-31040   Alabama   
..   ...                             ...                       ...       ...   
95   NaN  Brain and Other Nervous System               31010-31040  Maryland   
96   NaN  Brain and Other Nervous System               31010-31040  Maryland   
97   NaN  Brain and Other Nervous System               31010-31040  Maryland   
98   NaN  Brain and Other Nervous System               31010-31040  Maryland   
99   NaN  Brain and Other Nervous System               31010-31040  Maryland   

    States Code    Year  Year Code    Age Groups Age Groups Code   Sex  \
0           1.0  2019.0     2019.0   50-54 years           50-54  Male   
1           1.0  2019.0     2019.0   55-59 years           55-59  Male   
2           1.0  2019.0     2019.0  60-64 years            60-64  Male   
3           1.0  2019.0     2019.0   65-69 years           65-69  Male   
4           1.0  2019.0     2019.0   75-79 years           75-79  Male   
..          ...     ...        ...           ...             ...   ...   
95         24.0  2019.0     2019.0   55-59 years           55-59  Male   
96         24.0  2019.0     2019.0  60-64 years            60-64  Male   
97         24.0  2019.0     2019.0   65-69 years           65-69  Male   
98         24.0  2019.0     2019.0   70-74 years           70-74  Male   
99         24.0  2019.0     2019.0   75-79 years           75-79  Male   

   Sex Code  Count  Population  Crude Rate  
0         M   19.0    149277.0        12.7  
1         M   23.0    161482.0        14.2  
2         M   30.0    151504.0        19.8  
3         M   36.0    127679.0        28.2  
4         M   20.0     68256.0        29.3  
..      ...    ...         ...         ...  
95        M   29.0    206327.0        14.1  
96        M   25.0    182828.0        13.7  
97        M   29.0    143139.0        20.3  
98        M   33.0    112376.0        29.4  
99        M   28.0     73831.0        37.9  

[100 rows x 14 columns]

In [9]:
## Dropping Notes columns as we are not going to use it
df_2019=df_2019.drop(columns=['Notes'])

In [10]:
df_2019.shape

(12240, 13)

In [11]:
df_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12240 entries, 0 to 12239
Data columns (total 13 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Leading Cancer Sites       12178 non-null  object 
 1   Leading Cancer Sites Code  12178 non-null  object 
 2   States                     12178 non-null  object 
 3   States Code                12178 non-null  float64
 4   Year                       12178 non-null  float64
 5   Year Code                  12178 non-null  float64
 6   Age Groups                 12178 non-null  object 
 7   Age Groups Code            12178 non-null  object 
 8   Sex                        12178 non-null  object 
 9   Sex Code                   12178 non-null  object 
 10  Count                      12178 non-null  float64
 11  Population                 12178 non-null  float64
 12  Crude Rate                 12178 non-null  float64
dtypes: float64(6), object(7)
memory usage: 1.2+ MB

In [12]:
#Dropping all codes columns as we wont be using them

df_2019.drop(columns=['Leading Cancer Sites Code','States Code','Year Code','Age Groups Code'],inplace=True)

In [13]:
df_2019.shape

(12240, 9)

## Checking value counts to see the overall data

In [14]:
df_2019['Leading Cancer Sites'].value_counts(dropna=False)

Colon and Rectum                         987
Melanoma of the Skin                     950
Lung and Bronchus                        860
Non-Hodgkin Lymphoma                     808
Kidney and Renal Pelvis                  777
Leukemias                                742
Thyroid                                  704
Urinary Bladder, invasive and in situ    656
Breast                                   641
Pancreas                                 630
Oral Cavity and Pharynx                  630
Myeloma                                  485
Corpus Uteri                             445
Prostate                                 440
Brain and Other Nervous System           431
Stomach                                  430
Liver                                    395
Esophagus                                332
Ovary                                    317
Cervix Uteri                             243
Larynx                                   214
NaN                                       62
Gallbladde

In [15]:
df_2019['States'].value_counts(sort=True,dropna=False)

California              478
Texas                   463
New York                454
Florida                 447
Illinois                390
Ohio                    383
Pennsylvania            383
North Carolina          367
Georgia                 356
New Jersey              353
Michigan                347
Virginia                328
Washington              318
Tennessee               309
Massachusetts           309
Arizona                 308
Missouri                302
Wisconsin               298
Indiana                 296
Maryland                296
Colorado                280
Alabama                 279
Minnesota               278
Louisiana               275
South Carolina          272
Kentucky                272
Oklahoma                250
Oregon                  241
Connecticut             239
Iowa                    234
Arkansas                227
Mississippi             211
Kansas                  208
West Virginia           175
Utah                    174
Nebraska            

In [16]:
df_2019['Year'].value_counts(dropna=False)

2019.0    12178
NaN          62
Name: Year, dtype: int64

In [17]:
df_2019['Age Groups'].value_counts(dropna=False)

65-69 years     1404
60-64 years     1387
70-74 years     1381
75-79 years     1292
55-59 years     1266
80-84 years     1099
85+ years       1050
50-54 years     1011
45-49 years      775
40-44 years      534
35-39 years      396
30-34 years      262
25-29 years      145
20-24 years       64
NaN               62
1-4 years         37
15-19 years       32
10-14 years       22
5-9 years         21
Name: Age Groups, dtype: int64

In [18]:
df_2019['Sex'].value_counts(dropna=False)

Female    6265
Male      5913
NaN         62
Name: Sex, dtype: int64

In [19]:
df_2019['Sex Code'].value_counts(dropna=False)

F      6265
M      5913
NaN      62
Name: Sex Code, dtype: int64

In [20]:
df_2019['Count'].value_counts(ascending=True,dropna=False)

893.0       1
419.0       1
645.0       1
582.0       1
1004.0      1
         ... 
20.0      284
18.0      289
19.0      289
17.0      318
16.0      344
Name: Count, Length: 883, dtype: int64

In [21]:
df_2019['Count'].unique()

array([  19.,   23.,   30.,   36.,   20.,   17.,   22.,   24.,   33.,
         21.,   16.,   38.,   42.,   28.,   27.,   57.,   50.,   73.,
         74.,   95.,  134.,  164.,  169.,  137.,  113.,   67.,   53.,
         18.,   26.,   32.,   35.,   96.,  117.,  148.,  109.,   91.,
         68.,   51.,   25.,   52.,   29.,   31.,   37.,   59.,   39.,
         46.,   45.,   40.,   48.,   75.,   82.,   80.,   69.,   54.,
         47.,   65.,   72.,   55.,   70.,   44.,   34.,   84.,  100.,
         97.,   41.,   49.,  131.,  153.,  179.,  223.,  183.,  130.,
        114.,   81.,   99.,  176.,  213.,  199.,  170.,  125.,  122.,
         79.,  102.,  128.,  152.,   87.,  155.,  270.,  471.,  845.,
        923., 1057., 1069.,  935.,  755.,  638.,  646.,   58.,  112.,
        121.,  133.,   93.,   71.,   88.,   90.,  115.,   94.,   98.,
         61.,   76.,  129.,  273.,  449.,  593.,  716.,  835.,  750.,
        669.,  500.,  477.,   78.,  149.,  269.,  299.,  358.,  384.,
        337.,  250.,

Observation from the above analysis.
There are about 60 Nan values across the data.

## Mixed data types check

In [22]:
df_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12240 entries, 0 to 12239
Data columns (total 9 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Leading Cancer Sites  12178 non-null  object 
 1   States                12178 non-null  object 
 2   Year                  12178 non-null  float64
 3   Age Groups            12178 non-null  object 
 4   Sex                   12178 non-null  object 
 5   Sex Code              12178 non-null  object 
 6   Count                 12178 non-null  float64
 7   Population            12178 non-null  float64
 8   Crude Rate            12178 non-null  float64
dtypes: float64(4), object(5)
memory usage: 860.8+ KB


In [23]:
for col in df_2019.columns.tolist():
  weird = (df_2019[[col]].applymap(type) != df_2019[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_2019[weird]) > 0:
    print (col)

Leading Cancer Sites
States
Age Groups
Sex
Sex Code


## There are 5 columns with mixed data types. FIrst we will deal with missing values and check again if the difference is because of them.

In [24]:
#checking NULL(missing data) rows in columns
df_2019.isnull().sum()

Leading Cancer Sites    62
States                  62
Year                    62
Age Groups              62
Sex                     62
Sex Code                62
Count                   62
Population              62
Crude Rate              62
dtype: int64

In [25]:
## as all columns have NULL, we can check if all rows are null and then columnwise.

In [26]:
#dropping NaN rows
df_2019.dropna(inplace = True)

In [27]:
df_2019.shape

(12178, 9)

## 12238 rows reduced to 12178. 
That is our 60 rows of NULL value. Lets check mixed and missing values again.

In [28]:
#Missing values
df_2019.isnull().sum()

Leading Cancer Sites    0
States                  0
Year                    0
Age Groups              0
Sex                     0
Sex Code                0
Count                   0
Population              0
Crude Rate              0
dtype: int64

In [29]:
#mixed data columns
for col in df_2019.columns.tolist():
  weird = (df_2019[[col]].applymap(type) != df_2019[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_2019[weird]) > 0:
    print (col)

In [30]:
#checking duplicates

df_dups_2019 = df_2019[df_2019.duplicated()]

In [31]:
df_dups_2019

Empty DataFrame
Columns: [Leading Cancer Sites, States, Year, Age Groups, Sex, Sex Code, Count, Population, Crude Rate]
Index: []

In [32]:
#final 2019 clean data
df_2019.head(10)

Leading Cancer Sites   States    Year    Age Groups   Sex  \
0  Brain and Other Nervous System  Alabama  2019.0   50-54 years  Male   
1  Brain and Other Nervous System  Alabama  2019.0   55-59 years  Male   
2  Brain and Other Nervous System  Alabama  2019.0  60-64 years   Male   
3  Brain and Other Nervous System  Alabama  2019.0   65-69 years  Male   
4  Brain and Other Nervous System  Alabama  2019.0   75-79 years  Male   
5  Brain and Other Nervous System  Arizona  2019.0   50-54 years  Male   
6  Brain and Other Nervous System  Arizona  2019.0   55-59 years  Male   
7  Brain and Other Nervous System  Arizona  2019.0  60-64 years   Male   
8  Brain and Other Nervous System  Arizona  2019.0   65-69 years  Male   
9  Brain and Other Nervous System  Arizona  2019.0   70-74 years  Male   

  Sex Code  Count  Population  Crude Rate  
0        M   19.0    149277.0        12.7  
1        M   23.0    161482.0        14.2  
2        M   30.0    151504.0        19.8  
3        M   36.0    127679.0        28.2  
4        M   20.0     68256.0        29.3  
5        M   17.0    206772.0         8.2  
6        M   22.0    216293.0        10.2  
7        M   30.0    204086.0        14.7  
8        M   24.0    184808.0        13.0  
9        M   33.0    166440.0        19.8

## The data seems clean without NULL values , mixed data or Duplicates

In [33]:
#changing year to integer
df_2019['Year']=df_2019['Year'].astype('int')

In [34]:
df_2019.shape

(12178, 9)

In [35]:
df_2019.head(10)

Leading Cancer Sites   States  Year    Age Groups   Sex Sex Code  \
0  Brain and Other Nervous System  Alabama  2019   50-54 years  Male        M   
1  Brain and Other Nervous System  Alabama  2019   55-59 years  Male        M   
2  Brain and Other Nervous System  Alabama  2019  60-64 years   Male        M   
3  Brain and Other Nervous System  Alabama  2019   65-69 years  Male        M   
4  Brain and Other Nervous System  Alabama  2019   75-79 years  Male        M   
5  Brain and Other Nervous System  Arizona  2019   50-54 years  Male        M   
6  Brain and Other Nervous System  Arizona  2019   55-59 years  Male        M   
7  Brain and Other Nervous System  Arizona  2019  60-64 years   Male        M   
8  Brain and Other Nervous System  Arizona  2019   65-69 years  Male        M   
9  Brain and Other Nervous System  Arizona  2019   70-74 years  Male        M   

   Count  Population  Crude Rate  
0   19.0    149277.0        12.7  
1   23.0    161482.0        14.2  
2   30.0    151504.0        19.8  
3   36.0    127679.0        28.2  
4   20.0     68256.0        29.3  
5   17.0    206772.0         8.2  
6   22.0    216293.0        10.2  
7   30.0    204086.0        14.7  
8   24.0    184808.0        13.0  
9   33.0    166440.0        19.8

## Cleaning process 2018 data

In [36]:
#checking the dataypes and other cehcks on data frame
df_2018.shape

(12325, 14)

In [37]:
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12325 entries, 0 to 12324
Data columns (total 14 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Notes                      62 non-null     object 
 1   Leading Cancer Sites       12263 non-null  object 
 2   Leading Cancer Sites Code  12263 non-null  object 
 3   States                     12263 non-null  object 
 4   States Code                12263 non-null  float64
 5   Year                       12263 non-null  float64
 6   Year Code                  12263 non-null  float64
 7   Age Groups                 12263 non-null  object 
 8   Age Groups Code            12263 non-null  object 
 9   Sex                        12263 non-null  object 
 10  Sex Code                   12263 non-null  object 
 11  Count                      12263 non-null  float64
 12  Population                 12263 non-null  float64
 13  Crude Rate                 12263 non-null  flo

## Considering this is a similar set of data, we will drop Notes, Leading Cancer Sites Code,States Code ,Year Code and Age Group Code

In [38]:
#dropping columns
df_2018.drop(columns=['Notes','Leading Cancer Sites Code','States Code','Year Code','Age Groups Code'],inplace=True)

In [39]:
#verifying
df_2018.shape

(12325, 9)

In [40]:
#verifying
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12325 entries, 0 to 12324
Data columns (total 9 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Leading Cancer Sites  12263 non-null  object 
 1   States                12263 non-null  object 
 2   Year                  12263 non-null  float64
 3   Age Groups            12263 non-null  object 
 4   Sex                   12263 non-null  object 
 5   Sex Code              12263 non-null  object 
 6   Count                 12263 non-null  float64
 7   Population            12263 non-null  float64
 8   Crude Rate            12263 non-null  float64
dtypes: float64(4), object(5)
memory usage: 866.7+ KB


## Checking value counts to check NULL values and 

In [41]:
df_2018.isnull().sum()

Leading Cancer Sites    62
States                  62
Year                    62
Age Groups              62
Sex                     62
Sex Code                62
Count                   62
Population              62
Crude Rate              62
dtype: int64

In [42]:
#dropping NULL values
df_2018.dropna(inplace = True)

In [43]:
#verify
df_2018.shape

(12263, 9)

In [44]:
##this shows all 59 rows were empty rows.

In [45]:
#mixed data columns
for col in df_2018.columns.tolist():
  weird = (df_2018[[col]].applymap(type) != df_2018[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_2018[weird]) > 0:
    print (col)

In [46]:
#checking null again
df_2018.isnull().sum()

Leading Cancer Sites    0
States                  0
Year                    0
Age Groups              0
Sex                     0
Sex Code                0
Count                   0
Population              0
Crude Rate              0
dtype: int64

In [47]:
#checking duplicates
df_dups_2018 = df_2018[df_2018.duplicated()]

In [48]:
df_dups_2018

Empty DataFrame
Columns: [Leading Cancer Sites, States, Year, Age Groups, Sex, Sex Code, Count, Population, Crude Rate]
Index: []

### The data is clean without duplicates,mixed data type and NULL values

In [49]:
#changing year to integer
df_2018['Year']=df_2018['Year'].astype('int')

In [50]:
df_2018.shape

(12263, 9)

In [51]:
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 12263 entries, 0 to 12262
Data columns (total 9 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Leading Cancer Sites  12263 non-null  object 
 1   States                12263 non-null  object 
 2   Year                  12263 non-null  int64  
 3   Age Groups            12263 non-null  object 
 4   Sex                   12263 non-null  object 
 5   Sex Code              12263 non-null  object 
 6   Count                 12263 non-null  float64
 7   Population            12263 non-null  float64
 8   Crude Rate            12263 non-null  float64
dtypes: float64(3), int64(1), object(5)
memory usage: 958.0+ KB


In [52]:
df_2018.describe()

Year         Count    Population    Crude Rate
count  12263.0  12263.000000  1.226300e+04  12263.000000
mean    2018.0    122.640871  2.344209e+05     89.401517
std        0.0    240.223132  2.539047e+05    125.048592
min     2018.0     16.000000  3.525000e+03      1.100000
25%     2018.0     27.000000  7.266800e+04     18.300000
50%     2018.0     51.000000  1.576720e+05     41.300000
75%     2018.0    115.000000  2.900540e+05     95.000000
max     2018.0   4886.000000  1.614451e+06    920.500000

In [53]:
#final data frame for 2018
df_2018.head()

Leading Cancer Sites   States  Year    Age Groups     Sex  \
0  Brain and Other Nervous System  Alabama  2018  60-64 years   Female   
1  Brain and Other Nervous System  Alabama  2018   65-69 years  Female   
2  Brain and Other Nervous System  Alabama  2018   70-74 years  Female   
3  Brain and Other Nervous System  Alabama  2018   75-79 years  Female   
4  Brain and Other Nervous System  Alabama  2018   80-84 years  Female   

  Sex Code  Count  Population  Crude Rate  
0        F   23.0    168342.0        13.7  
1        F   21.0    145302.0        14.5  
2        F   23.0    117686.0        19.5  
3        F   19.0     84846.0        22.4  
4        F   18.0     58233.0        30.9

## Appending data frames into one

In [54]:
#checking and making sure both data sets have same format
df_2019.shape

(12178, 9)

In [55]:
df_2018.shape

(12263, 9)

In [56]:
#creating frame
frames=[df_2018,df_2019]

In [57]:
frames

[                        Leading Cancer Sites   States  Year    Age Groups  \
 0             Brain and Other Nervous System  Alabama  2018  60-64 years    
 1             Brain and Other Nervous System  Alabama  2018   65-69 years   
 2             Brain and Other Nervous System  Alabama  2018   70-74 years   
 3             Brain and Other Nervous System  Alabama  2018   75-79 years   
 4             Brain and Other Nervous System  Alabama  2018   80-84 years   
 ...                                      ...      ...   ...           ...   
 12258  Urinary Bladder, invasive and in situ  Wyoming  2018   65-69 years   
 12259  Urinary Bladder, invasive and in situ  Wyoming  2018   70-74 years   
 12260  Urinary Bladder, invasive and in situ  Wyoming  2018   75-79 years   
 12261  Urinary Bladder, invasive and in situ  Wyoming  2018   80-84 years   
 12262  Urinary Bladder, invasive and in situ  Wyoming  2018     85+ years   
 
           Sex Sex Code  Count  Population  Crude Rate  
 0   

In [58]:
#concatenating both data frames into one
df_final_1819=pd.concat(frames)

In [59]:
df_final_1819

Leading Cancer Sites     States  Year    Age Groups  \
0             Brain and Other Nervous System    Alabama  2018  60-64 years    
1             Brain and Other Nervous System    Alabama  2018   65-69 years   
2             Brain and Other Nervous System    Alabama  2018   70-74 years   
3             Brain and Other Nervous System    Alabama  2018   75-79 years   
4             Brain and Other Nervous System    Alabama  2018   80-84 years   
...                                      ...        ...   ...           ...   
12173  Urinary Bladder, invasive and in situ  Wisconsin  2019   65-69 years   
12174  Urinary Bladder, invasive and in situ  Wisconsin  2019   70-74 years   
12175  Urinary Bladder, invasive and in situ  Wisconsin  2019   75-79 years   
12176  Urinary Bladder, invasive and in situ  Wisconsin  2019   80-84 years   
12177  Urinary Bladder, invasive and in situ  Wisconsin  2019     85+ years   

          Sex Sex Code  Count  Population  Crude Rate  
0      Female        F   23.0    168342.0        13.7  
1      Female        F   21.0    145302.0        14.5  
2      Female        F   23.0    117686.0        19.5  
3      Female        F   19.0     84846.0        22.4  
4      Female        F   18.0     58233.0        30.9  
...       ...      ...    ...         ...         ...  
12173  Female        F   41.0    173144.0        23.7  
12174  Female        F   76.0    133024.0        57.1  
12175  Female        F   69.0     94842.0        72.8  
12176  Female        F   53.0     67245.0        78.8  
12177  Female        F   69.0     82068.0        84.1  

[24441 rows x 9 columns]

In [60]:
df_final_1819.describe()

Year         Count    Population    Crude Rate
count  24441.000000  24441.000000  2.444100e+04  24441.000000
mean    2018.498261    123.294505  2.354541e+05     88.903699
std        0.500007    243.495846  2.541393e+05    124.574493
min     2018.000000     16.000000  3.525000e+03      1.100000
25%     2018.000000     27.000000  7.383100e+04     18.200000
50%     2018.000000     51.000000  1.582010e+05     41.100000
75%     2019.000000    116.000000  2.915870e+05     94.200000
max     2019.000000   5263.000000  1.614451e+06    920.500000

In [61]:
#renaming Sex and Sex code to Gender
df_final_1819.rename(columns={'Sex': 'Gender','Sex Code':'Gender Code'},inplace=True)

In [62]:
df_final_1819

Leading Cancer Sites     States  Year    Age Groups  \
0             Brain and Other Nervous System    Alabama  2018  60-64 years    
1             Brain and Other Nervous System    Alabama  2018   65-69 years   
2             Brain and Other Nervous System    Alabama  2018   70-74 years   
3             Brain and Other Nervous System    Alabama  2018   75-79 years   
4             Brain and Other Nervous System    Alabama  2018   80-84 years   
...                                      ...        ...   ...           ...   
12173  Urinary Bladder, invasive and in situ  Wisconsin  2019   65-69 years   
12174  Urinary Bladder, invasive and in situ  Wisconsin  2019   70-74 years   
12175  Urinary Bladder, invasive and in situ  Wisconsin  2019   75-79 years   
12176  Urinary Bladder, invasive and in situ  Wisconsin  2019   80-84 years   
12177  Urinary Bladder, invasive and in situ  Wisconsin  2019     85+ years   

       Gender Gender Code  Count  Population  Crude Rate  
0      Female           F   23.0    168342.0        13.7  
1      Female           F   21.0    145302.0        14.5  
2      Female           F   23.0    117686.0        19.5  
3      Female           F   19.0     84846.0        22.4  
4      Female           F   18.0     58233.0        30.9  
...       ...         ...    ...         ...         ...  
12173  Female           F   41.0    173144.0        23.7  
12174  Female           F   76.0    133024.0        57.1  
12175  Female           F   69.0     94842.0        72.8  
12176  Female           F   53.0     67245.0        78.8  
12177  Female           F   69.0     82068.0        84.1  

[24441 rows x 9 columns]

## Creating some flags for further analysis.

In [63]:
#State to regional categorization North-East
df_final_1819.loc[df_final_1819['States'].isin(['New York','Connecticut','Maine','New Hampshire',
       'New Jersey','Pennsylvania','Vermont','Massachusetts','Rhode Island']), 'Region'] = 'North-East'
#updating mid-west
df_final_1819.loc[df_final_1819['States'].isin(['Illinois', 'Indiana','Iowa','Kansas','Michigan', 'Minnesota','Missouri',
                                           'Nebraska','North Dakota','Ohio','South Dakota','Wisconsin']), 'Region'] = 'Mid-West'
#updating south
df_final_1819.loc[df_final_1819['States'].isin(['Alabama','Arkansas','Delaware', 'District of Columbia',
       'Florida', 'Georgia', 'Kentucky', 'Louisiana','Maryland','Mississippi','North Carolina','Tennessee',
       'Texas','Virginia', 'West Virginia','South Carolina','Oklahoma']), 'Region'] = 'South'
#updating west
df_final_1819.loc[df_final_1819['States'].isin(['Alaska', 'Arizona','California','Hawaii', 'Idaho','Oregon','New Mexico', 
       'Colorado','Nevada','Washington','Wyoming','Montana', 'Utah']), 'Region'] = 'West'

In [64]:
df_final_1819['Region'].value_counts(dropna=False)

South         9314
Mid-West      6047
West          4725
North-East    4355
Name: Region, dtype: int64

1-14 : Pediatric group
15-29 :Young age group
30-49 : Adult age group
50-64 : Middle age group
65+ : Elderly group

In [65]:
#age group flagging
df_final_1819.loc[df_final_1819['Age Groups'].isin(['1-4 years','5-9 years','10-14 years']), 'Age_Flag'] = 'Pediatric group'
df_final_1819.loc[df_final_1819['Age Groups'].isin(['15-19 years','20-24 years','25-29 years']), 'Age_Flag'] = 'Young age group'
df_final_1819.loc[df_final_1819['Age Groups'].isin(['30-34 years','35-39 years','40-44 years','45-49 years']), 'Age_Flag'] = 'Adult age group'
df_final_1819.loc[df_final_1819['Age Groups'].isin(['50-54 years','55-59 years','60-64 years ']), 'Age_Flag'] = 'Middle age group'
df_final_1819.loc[df_final_1819['Age Groups'].isin(['65-69 years','70-74 years','75-79 years','80-84 years','85+ years']), 'Age_Flag'] = 'Elderly group'


In [67]:
df_final_1819['Age_Flag'].value_counts(dropna=False)

Elderly group       12470
Middle age group     7355
Adult age group      3952
Young age group       495
Pediatric group       169
Name: Age_Flag, dtype: int64

In [68]:
#age group numeric flagging

df_final_1819.loc[df_final_1819['Age Groups'].isin(['1-4 years']),"Age_max"]=4
df_final_1819.loc[df_final_1819['Age Groups'].isin(['5-9 years']),"Age_max"]=9
df_final_1819.loc[df_final_1819['Age Groups'].isin(['10-14 years']),"Age_max"]=14
df_final_1819.loc[df_final_1819['Age Groups'].isin(['15-19 years']),"Age_max"]=19
df_final_1819.loc[df_final_1819['Age Groups'].isin(['20-24 years']),"Age_max"]=24
df_final_1819.loc[df_final_1819['Age Groups'].isin(['25-29 years']),"Age_max"]=29
df_final_1819.loc[df_final_1819['Age Groups'].isin(['30-34 years']),"Age_max"]=34
df_final_1819.loc[df_final_1819['Age Groups'].isin(['35-39 years']),"Age_max"]=39
df_final_1819.loc[df_final_1819['Age Groups'].isin(['40-44 years']),"Age_max"]=44
df_final_1819.loc[df_final_1819['Age Groups'].isin(['45-49 years']),"Age_max"]=49
df_final_1819.loc[df_final_1819['Age Groups'].isin(['50-54 years']),"Age_max"]=54
df_final_1819.loc[df_final_1819['Age Groups'].isin(['55-59 years']),"Age_max"]=59
df_final_1819.loc[df_final_1819['Age Groups'].isin(['60-64 years ']),"Age_max"]=64
df_final_1819.loc[df_final_1819['Age Groups'].isin(['65-69 years']),"Age_max"]=69
df_final_1819.loc[df_final_1819['Age Groups'].isin(['70-74 years']),"Age_max"]=74
df_final_1819.loc[df_final_1819['Age Groups'].isin(['75-79 years']),"Age_max"]=79
df_final_1819.loc[df_final_1819['Age Groups'].isin(['80-84 years']),"Age_max"]=84
df_final_1819.loc[df_final_1819['Age Groups'].isin(['85+ years']),"Age_max"]=89

In [69]:
df_final_1819['Age_max'].value_counts(dropna=False)

69.0    2818
64.0    2765
74.0    2759
79.0    2574
59.0    2553
84.0    2209
89.0    2110
54.0    2037
49.0    1552
44.0    1085
39.0     793
34.0     522
29.0     299
24.0     130
4.0       79
19.0      66
14.0      46
9.0       44
Name: Age_max, dtype: int64

In [71]:
df_final_1819.head(10)

Leading Cancer Sites   States  Year    Age Groups  Gender  \
0  Brain and Other Nervous System  Alabama  2018  60-64 years   Female   
1  Brain and Other Nervous System  Alabama  2018   65-69 years  Female   
2  Brain and Other Nervous System  Alabama  2018   70-74 years  Female   
3  Brain and Other Nervous System  Alabama  2018   75-79 years  Female   
4  Brain and Other Nervous System  Alabama  2018   80-84 years  Female   
5  Brain and Other Nervous System  Arizona  2018   55-59 years  Female   
6  Brain and Other Nervous System  Arizona  2018  60-64 years   Female   
7  Brain and Other Nervous System  Arizona  2018   65-69 years  Female   
8  Brain and Other Nervous System  Arizona  2018   70-74 years  Female   
9  Brain and Other Nervous System  Arizona  2018   75-79 years  Female   

  Gender Code  Count  Population  Crude Rate Region          Age_Flag  Age_max  
0           F   23.0    168342.0        13.7  South  Middle age group     64.0  
1           F   21.0    145302.0        14.5  South     Elderly group     69.0  
2           F   23.0    117686.0        19.5  South     Elderly group     74.0  
3           F   19.0     84846.0        22.4  South     Elderly group     79.0  
4           F   18.0     58233.0        30.9  South     Elderly group     84.0  
5           F   21.0    227767.0         9.2   West  Middle age group     59.0  
6           F   23.0    224349.0        10.3   West  Middle age group     64.0  
7           F   22.0    208318.0        10.6   West     Elderly group     69.0  
8           F   20.0    178478.0        11.2   West     Elderly group     74.0  
9           F   17.0    127735.0        13.3   West     Elderly group     79.0

## Exporting data set to prepared data

In [72]:
df_final_1819.to_csv(os.path.join(pathk,'Prepared Data','2018_19_cancer_incidents.csv'))